power_grid_model_c/power_grid_model/include/power_grid_model/math_solver/measured_values.hpp

```diff
@@ -423,70 +453,48 @@ template <symmetry_tag sym> class MeasuredValues {
     template <bool only_magnitude = false>
     static VoltageSensorCalcParam<sym> combine_measurements(std::vector<VoltageSensorCalcParam<sym>> const& data,
                                                             IdxRange const& sensors) {
-        double accumulated_inverse_variance{};
-        ComplexValue<sym> accumulated_value{};
-        for (auto pos : sensors) {
-            auto const& measurement = data[pos];
-            auto const inv_variance = 1.0 / measurement.variance;
-
-            accumulated_inverse_variance += inv_variance;
-            if constexpr (only_magnitude) {
-                ComplexValue<sym> abs_value = piecewise_complex_value<sym>(DoubleComplex{0.0, nan});
-                abs_value += detail::cabs_or_real<sym>(measurement.value);
-                accumulated_value += abs_value * inv_variance;
-            } else {
-                // accumulate value
-                accumulated_value += measurement.value * inv_variance;
-            }
-        }
-
-        if (!is_normal(accumulated_inverse_variance)) {
-            return VoltageSensorCalcParam<sym>{accumulated_value, std::numeric_limits<double>::infinity()};
+        auto complex_measurements = sensors | std::views::transform([&data](Idx pos) -> auto& { return data[pos]; });
+        if constexpr (only_magnitude) {
```

Comment:
I had thought this tiny only_magnitude param would be done away with somehow.
We can move this logic to accumulate too right? With an additional if constexpr statement for UniformCompelxRandVar. with this logic of isnan, then cast to UniformRealRandVar. (This logic can be a separate function only_magnitude_cast)

Reply:
the problem lies in the fact that we chose to model the magnitude only voltage sensor calc param as a full uniform complex random variable in the first place. the decision to actually model the magnitude separately is a choice made in the measured values. the logic on how to handle magnitude-only measurements is already moved to `statistics.hpp` but i don't think we should move the logic on how to interpret magnitude-only measurements there as well (or maybe to a separate function with a separate name).

What do you think?

Reply:
Alright. That makes sense.
Then what if we move the check directly to combine measurements from process_voltage_measurements?
We have an additional logic for current sensors now too.
This too can remove the only_magnitude right?

Reply:
yes, but then we would have to have 2 lists of voltage measurements and a significant refactoring of the entire measured values.

Reply:
cfr. offline discussion: I will create a new function `statistics::combine_magnitude` with only the overload for `UniformComplexRandVar` implemented (`PolarComplexRandVar` would also be possible but since we have no other overloads for the `PolarComplexRandVar` version anyways, let's not do that yet)